### PR TITLE
add a check for dependencies provided via api plugins

### LINKED ISSUE #4047: Scan for dependencies provided by library plugins
### What feature do you want to see added?

When a new plugin declares a dependency to a java library that is provided by a library plugin, it should be reported that the dependency must be changed to the library plugin.

### Upstream changes

_No response_

### PR DESCRIPTION
**Pull request by mawinter69**
Request to change dependencies that are available via api plugins

Uses a simple text file for defining which dependencies should be checked and what should be used instead. Additional comments can be added.

fixes #4047

### CODE REVIEW

**Comment by mawinter69** [comment]
@NotMyFault 
I haven't added several api plugins around databases, e.g. https://plugins.jenkins.io/mysql-api/
Should those also be added maybe?
Probably other candidates are all the amazon sdk plugins

File: banned-dependencies.lst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+org.apache.commons:commons-lang3;io.jenkins.plugins:commons-lang3-api
+org.apache.commons:commons-lang;io.jenkins.plugins:commons-lang3-api
+org.apache.commons:commons-text;io.jenkins.plugins:commons-text-api
+org.apache.commons:commons-compress;io.jenkins.plugins:commons-compress-api
+org.apache.commons:commons-math3;io.jenkins.plugins:commons-math3-api
+
+org.json:json;io.jenkins.plugins:json-api
+com.jayway.jsonpath:json-path;io.jenkins.plugins:json-path-api
+com.google.code.gson:gson;io.jenkins.plugins:gson-api
+com.fasterxml.jackson.core:jackson-core;org.jenkins-ci.plugins:jackson2-api
```
Comment (mawinter69) [inline banned-dependencies.lst:11]: My assumption is that jackson-core is always included.

**Comment by mawinter69** [comment]
Other candidates:
https://plugins.jenkins.io/github-api/
https://plugins.jenkins.io/gitlab-api/
https://plugins.jenkins.io/eddsa-api/
https://plugins.jenkins.io/docker-java-api/
https://plugins.jenkins.io/kubernetes-client-api/
https://plugins.jenkins.io/commons-httpclient3-api/
https://plugins.jenkins.io/azure-sdk/

File: banned-dependencies.lst (hunk 1 above)
Comment (timja) [reply banned-dependencies.lst:11]: It can often come as a dependency of another Jackson one but we can see how it goes

Comment (mawinter69) [reply banned-dependencies.lst:11]: Another idea would be to ban the complete groupIds, e.g. `com.fasterxml.jackson.core` , `com.fasterxml.jackson.datatype`, `com.fasterxml.jackson.module` and `com.fasterxml.jackson.dataformat`

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Thanks!

**Pull request merged**